OSPSuite.Presentation.Importer/Core/IUnformattedData.cs

```diff
@@ -0,0 +1,127 @@
+      IEnumerable<IEnumerable<string>> GetRows(Func<IEnumerable<string>, bool> filter);
+
+      void AddRow(IEnumerable<string> row);
+      void AddColumn(string columnName, int columnIndex);
+      DataTable AsDataTable();
+      UnformattedDataRow GetDataRow(int index);
+   }
+
+   public class UnformattedData : IUnformattedData
+   {
+      private readonly List<List<string>> _rawDataTable = new List<List<string>>();
+
+      protected Cache<string, ColumnDescription> _headers =
+         new Cache<string, ColumnDescription>(); //we have to ensure headers and RawData sizes match
+
+      private IEnumerable<string> getColumn(int columnIndex)
+      {
+         return _rawDataTable.Select(column => column[columnIndex]).ToList();
+      }
+
+      public void AddColumn(string columnName, int columnIndex) //it seems to me there is little sense in adding column after column
+         //the list of headers is somehow the definition of the table
+      {
+         if (_headers.Keys.Contains(columnName))
+            columnName = Guid.NewGuid().ToString();
+         _headers.Add(columnName, new ColumnDescription(columnIndex));
+      }
+
+      public void CalculateColumnDescription(List<ColumnDescription.MeasurementLevel> levels)
+      {
+         _headers.Each(header =>
+         {
+            if (header.Level == ColumnDescription.MeasurementLevel.NotSet)
+               header.Level = levels[header.Index];
+         });
+      }
+
+      public void AddRow( IEnumerable<string> row)
+      {
+         var rowList = row.ToList();
+
+         //could it actually 
+         if (_headers.Count > rowList.Count)
+         {
+            for ( var i = rowList.Count; i < _headers.Count; i++  )
+               rowList.Add("");
+         }
+         if (rowList.Count > _headers.Count)
+         {
+            rowList = rowList.GetRange(0, _headers.Count);
+         }
+
+         _rawDataTable.Add(rowList);
+
+         foreach (var header in _headers)
+         {
+            if (header.Level != ColumnDescription.MeasurementLevel.Discrete) continue;
+            if (!header.ExistingValues.Contains(rowList.ElementAt(header.Index)))
+               header.ExistingValues.Add(rowList.ElementAt(header.Index));
+         }
+      }
+
+
+      public IEnumerable<string> GetColumn(string columnName)
+      {
+         return getColumn(_headers[columnName].Index);
+      }
+
+      public IEnumerable<IEnumerable<string>> GetRows(Func<IEnumerable<string>, bool> filter)
+      {
+         return _rawDataTable.Where(filter);
+      }
+
+      public DataTable AsDataTable()
+      {
+         var resultTable = new DataTable();
+
+         // Add columns.
+         foreach (var header in _headers.Keys)
+         {
+            resultTable.Columns.Add(header, typeof(string));
+         }
+
+         foreach (var itemList in _rawDataTable)
+         {
+            resultTable.Rows.Add(itemList.ToArray()); //TODO Resharper
```

Comment:
TODO here

Reply:
There is a resharper warning in this position that we discussed and thought we should ignore. We can discuss this on Monday/Wednesday if you want

Reply:
Hum. I think I had weird runtime crash by ignoring this warning.

Reply:
You could always have, let us say you have a string list and send it to a method expecting a list of objects. In the second object you could turn some values into something more. If later you try to reuse the list as strings your app will crash. Here we send the list to the Rows.Add and forget about the list so, not problem should arise. There could be ways to avoid the warning but does it make sense? As long as we are clear of the implications I think it is ok.

Reply:
resharper warning disabled with comment